pystac_client/stac_api_io.py

```diff
@@ -263,20 +265,47 @@ def get_pages(
                 (link for link in page.get("links", []) if link["rel"] == "next"), None
             )
 
-    def assert_conforms_to(self, conformance_class: ConformanceClasses) -> None:
-        """Raises a :exc:`NotImplementedError` if the API does not publish the given
+    def assert_conforms_to(self, *conformance_classes: ConformanceClasses) -> None:
+        """Raises a :exc:`DoesNotConformTo` if the API does not publish the given
         conformance class. This method only checks against the ``"conformsTo"``
         property from the API landing page and does not make any additional
         calls to a ``/conformance`` endpoint even if the API provides such an endpoint.
 
         Args:
-            conformance_class: The ``ConformanceClasses`` key to check conformance
-            against.
+            conformance_classes: The ``ConformanceClasses`` key to check conformance
+            against. If any match, this passes.
         """
-        if not self.conforms_to(conformance_class):
-            raise NotImplementedError(f"{conformance_class} not supported")
+        with warnings.catch_warnings(action="error", category=DoesNotConformTo):
+            self.conforms_to(*conformance_classes)
 
-    def conforms_to(self, conformance_class: ConformanceClasses) -> bool:
+    def conforms_to(self, *conformance_classes: ConformanceClasses) -> bool:
+        """Whether the API conforms to the given standards.
+
+        If False this method raises a ``DoesNotConformTo`` warning.
+        This method only checks against the ``"conformsTo"``
+        property from the API landing page and does not make any additional
+        calls to a ``/conformance`` endpoint even if the API provides such an endpoint.
+
+        Args:
+            conformance_classes: The ``ConformanceClasses`` key to check conformance
+            against. If any match, this returns True and does not warn.
+        Return:
+            bool: Indicates if the API conforms to the given spec or URI.
+
+        """
+        if any(map(self._conforms_to, conformance_classes)):
+            return True
+        else:
+            warnings.warn(
+                (
+                    "Catalog does not conform to "
+                    f"{', '.join(c.name for c in conformance_classes)}"
+                ),
+                category=DoesNotConformTo,
+            )
```

Comment:
Is this the correct place to raise the warning? I ask, because there could be times that we call `conforms_to()` in a non-warning scenario (e.g. maybe we just want to report info about a API server). It feels better (to me) to raise this warning at the point of call, not the point of execution.

Reply:
Yeah I struggled with that. On the one hand I want to make it very easy for developers to not forget to add warnings, on the other hand I did end up using the private `_conforms_to` in a few places precisely to avoid the warning.